src/main/java/server/HTTPServer.java

```diff
@@ -1,37 +1,36 @@
 package server;
 
 import java.io.IOException;
-import java.util.ArrayList;
 
 public class HTTPServer implements Application {
 
     private final String protocol;
+    private final RequestParsable requestParser;
 
-    public HTTPServer() {
+    public HTTPServer(RequestParsable requestParser) {
         this.protocol = "HTTP/1.1";
+        this.requestParser = requestParser;
     }
 
     @Override
     public void call(ReadableWriteable readerWriter) throws IOException {
-        ArrayList<String> request = readRequest(readerWriter);
-        String path = getPath(request);
-        Response response = route(path);
+        Request request = requestParser.call(readRequest(readerWriter));
+        Response response = route(getPath(request));
         response.write(readerWriter);
     }
 
-    private ArrayList<String> readRequest(ReadableWriteable readerWriter) throws IOException {
-        String line = readerWriter.readLine();
-        ArrayList<String> request = new ArrayList<>();
+    private String getPath(Request request) {
```

Comment:
I'm not sure where to put this method. 

I made the Request record's requestLine property encompass the entire line (method, path, and protocol) for ease of use, but then it's a bit cumbersome for collaborators to pull out pieces of it. 

I'm considering changing Request to a regular class and adding some getters for each part of the request line. Any thoughts?